### ISSUE TITLE
When my config content is array, I get a {}

### ISSUE THREAD

**Issue by Muluk-m** [antfu/unconfig#26]
### Describe the bug

It seems that the reason is that array is not supported when using defu for config merge
<img width="688" alt="image" src="https://github.com/antfu/unconfig/assets/69963532/f5417983-6c0d-473f-90f3-d3d3c5695ea5">

### Reproduction

1. [build.config returns an array.](https://stackblitz.com/edit/stackblitz-starters-a8aypj?file=index.js)
2. In terminal, execute `npm run dev `

### System Info

```Shell
System:
    OS: macOS 13.3.1
    CPU: (8) x64 Apple M1 Pro
    Memory: 21.34 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.20.0 - ~/.nvm/versions/node/v16.20.0/bin/node
    Yarn: 1.22.17 - /opt/homebrew/bin/yarn
    npm: 8.19.4 - ~/.nvm/versions/node/v16.20.0/bin/npm
    pnpm: 8.15.4 - ~/.nvm/versions/node/v16.20.0/bin/pnpm
  Browsers:
    Chrome: 123.0.6312.87
    Safari: 16.4
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by antfu** [comment]
We temporarily closed this due to the lack of enough information. We could not identify whether it was a bug or a userland misconfiguration with the given info.
Please provide a [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example) to reopen the issue.
Thanks.

> [Why reproduction is required](https://antfu.me/posts/why-reproductions-are-required)

**Comment by Muluk-m** [comment]
> We temporarily closed this due to the lack of enough information. We could not identify whether it was a bug or a userland misconfiguration with the given info. Please provide a [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example) to reopen the issue. Thanks.
> 
> > [Why reproduction is required](https://antfu.me/posts/why-reproductions-are-required)

https://stackblitz.com/edit/stackblitz-starters-a8aypj?file=package.json

**Comment by antfu** [comment]
Fixed 84bea2a57d52fa72e4da3a4b7d969106dc36b1c2
